lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -63,9 +73,48 @@ def _get_config(self, config_dict):
                                config_dict['ovs_internal_sampling_port_number'],
             internal_sampling_fwd_tbl=
                               config_dict['ovs_internal_sampling_fwd_tbl_number'],
+            multi_tunnel_flag=multi_tunnel_flag,
 
         )
 
+    def _get_gtp_port_no(self, port_name:str):
+        output = subprocess.check_output("sudo ovsdb-client dump Interface name ofport",
+                                          shell=True,).decode("utf-8")
+        for line in output.split('\n'):
+            if port_name in line:
+                port_info = line.split()
+                return int(port_info[1])
+
+    def _ovs_multi_tunnel_init(self):
+        try:
+            subprocess.check_output("sudo ovs-vsctl list Open_vSwitch | grep gtpu",
+                                     shell=True,)
+            self.ovs_gtp_type = "gtpu"
+        except subprocess.CalledProcessError:
+            self.ovs_gtp_type = "gtp"
+
+    def _ip_addr_to_gtp_port_name(self, enodeb_ip_addr:str):
+        ip_no = hex(int(ipaddress.ip_address(enodeb_ip_addr)))
+        buf = "g_{}".format(ip_no[2:])
+        return buf
+
+    def _find_gtp_port_no(self, enodeb_ip_addr:str):
+        if not self.config.multi_tunnel_flag:
+            return self.config.gtp_port
+
+        port_name = self._ip_addr_to_gtp_port_name(enodeb_ip_addr)
+        add_cmd = "sudo ovs-vsctl --may-exist add-port gtp_br0 {} \
```

Comment:
Can you use RYU lib to create the tunnel port rather than cmdline.

Reply:
ok..Sure. We will do it using RYU lib.

Reply:
Changes done. Please review it.